File: iteration_1/src/Staff.java
import java.util.ArrayList;

public class Staff extends Person {
    private String reputation;
    private ArrayList<TimeInterval> officeHours;
    private int salary;
    private String employmentStatus;
    
    public Staff(String personName, String personSurname, String username, String password,
            String reputation, ArrayList<TimeInterval> officeHours, int salary, String employmentStatus) {
        super(personName, personSurname, username, password);
        this.reputation = reputation;
        this.officeHours = officeHours;
        this.salary = salary;
        this.employmentStatus = employmentStatus;
        this.officeHours = officeHours;
    }

    public void setOfficeHours(ArrayList<TimeInterval> officeHours) {
        this.officeHours = officeHours;
    }

    public Staff() {
        
    }

    public String getReputation() {
        return this.reputation;
    }

    public void setReputation(String reputation) {
        this.reputation = reputation;
    }

    public ArrayList<TimeInterval> getOfficeHours() {
        return this.officeHours;
    }
    public boolean addOfficeHours(TimeInterval officeHour){
        return officeHours.add(officeHour);
    }
    public boolean deleteOfficeHours(TimeInterval officeHour){
        return officeHours.remove(officeHour);
    }

    public int getSalary() {
        return this.salary;
    }

    public void setSalary(int salary) {
        this.salary = salary;
    }

    public String getEmploymentStatus() {
        return employmentStatus;
    }

    public void setEmploymentStatus(String employmentStatus) {
        this.employmentStatus = employmentStatus;
    }
}

File: python_iteration/models/Staff.py
from Person import Person


class Staff(Person):
    def __init__(
        self,
        person_name,
        person_surname,
        username,
        password,
        reputation,
        office_hours,
        salary,
        employment_status,
    ):
        super().__init__(person_name, person_surname, username, password)
        self.__reputation = reputation
        self.__office_hours = office_hours
        self.__salary = salary
        self.__employment_status = employment_status

    @property
    def reputation(self):
        return self.__reputation

    @reputation.setter
    def reputation(self, reputation):
        self.__reputation = reputation

    @property
    def office_hours(self):
        return self.__office_hours

    @office_hours.setter
    def office_hours(self, office_hours):
        self.__office_hours = office_hours

    def add_office_hour(self, office_hour):
        self.__office_hours.append(office_hour)
        return True

    def delete_office_hour(self, office_hour):
        if office_hour in self.__office_hours:
            self.__office_hours.remove(office_hour)
            return True
        return False

    @property
    def salary(self):
        return self.__salary

    @salary.setter
    def salary(self, salary):
        self.__salary = salary

    @property
    def employment_status(self):
        return self.__employment_status

    @employment_status.setter
    def employment_status(self, employment_status):
        self.__employment_status = employment_status
